odbc/src/api/functions.rs

```diff
@@ -1211,6 +1226,20 @@ fn sql_free_handle(handle_type: HandleType, handle: *mut MongoHandle) -> Result<
                     .as_connection()
                     .ok_or(ODBCError::InvalidHandleType(HANDLE_MUST_BE_CONN_ERROR))?
             };
+            // We must take the connection attributes in the mongo_connection and put them back
+            // in the connection attributes for the conn
+            if conn.has_mongo_connection() {
+                let mongo_connection_attributes = conn
+                    .mongo_connection
+                    .read()
+                    .unwrap()
+                    .as_ref()
+                    .unwrap()
+                    .attributes
+                    .clone();
+                let mut connection_attributes = conn.attributes.write().unwrap();
+                *connection_attributes = Some(mongo_connection_attributes);
```

Comment:
```suggestion
                let mongo_connection_attributes = conn
                    .mongo_connection
                    .read()
                    .unwrap()
                    .as_ref()
                    .unwrap()
                    .attributes
                    .clone();
                let mut connection_attributes = conn.attributes.write().unwrap();
                *connection_attributes = Some(mongo_connection_attributes);
```

This clone isn't necessary.

Reply:
this is dead code I thought I had removed. It's gone now!